crates/flox/src/commands/mod.rs

```diff
@@ -469,7 +471,46 @@ impl EnvironmentSelect {
                             if activated == current_dir {
                                 Self::open_env_pointer(flox, &current_dir, current_dir_pointer)
                             } else {
-                                todo!("needs design");
+                                let activated_pointer = EnvironmentPointer::open(&activated)?;
+                                let message = "Do you want to install to the current directory's flox environment or the current active flox environment?";
+                                let current_description = hacky_environment_description(
+                                    &current_dir,
+                                    &current_dir_pointer,
+                                )?;
+                                let activated_description =
+                                    hacky_environment_description(&activated, &activated_pointer)?;
+                                if Dialog::can_prompt() {
+                                    let dialog = Dialog {
+                                        message,
+                                        help_message: None,
+                                        typed: Select {
+                                            options: vec![
+                                                format!(
+                                                    "current directory's flox environment [{current_description}]",
+                                                ),
+                                                format!(
+                                                    "current active flox environment [{activated_description}]",
+                                                ),
+                                            ],
+                                        },
+                                    };
+                                    let (index, _) = dialog.raw_prompt()?;
+                                    match index {
+                                        0 => Self::open_env_pointer(
+                                            flox,
+                                            &current_dir,
+                                            current_dir_pointer,
+                                        ),
+                                        1 => Self::open_env_pointer(
+                                            flox,
+                                            &activated,
+                                            activated_pointer,
+                                        ),
+                                        _ => unreachable!(),
+                                    }
+                                } else {
+                                    Err(anyhow!("can't determine whether to use {current_description} or {activated_description}; specify an environment using --dir or --remote"))?
+                                }
```

Comment:
suggestion: move this into `Self::open_interactively`

Lets try to flatten the control flow a little.

*maybe add an early return for the `! use_activated` case, while we're at (non-blocking)*

Reply:
Split it out. I moved it outside `EnvironmentSelect` because I'll need to use it for search and show